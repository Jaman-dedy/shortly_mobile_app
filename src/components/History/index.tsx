import React from 'react';
import {ScrollView} from 'react-native';

import {Box, Text} from '@app/assets/style/theme';

import BoxComponent from '@app/components/Box';
import {IState} from '@app/types';
import PlaceHolder from '@app/components/common/PlaceHolder';
import EmptyContent from '@app/components/common/EmptyContent';

type Props = {
  shortenLinkState: IState;
  handleDeleteLink: (code: string) => void;
};

const History = ({shortenLinkState, handleDeleteLink}: Props) => {
  return (
    <Box flex={3} backgroundColor="offBackground" marginTop="md">
      <Text variant="paragraph">Your Link History</Text>
      <ScrollView>
        <Box marginTop="md">
          {shortenLinkState?.loading ? (
            <>
              <PlaceHolder />
              <PlaceHolder />
            </>
          ) : shortenLinkState?.result?.length ? (
            shortenLinkState?.result.map(
              (link: Record<string, string>, index: number) => (
                <BoxComponent
                  key={index}
                  link={link}
                  handleDeleteLink={handleDeleteLink}
                />
              ),
            )
          ) : (
            <>
              <EmptyContent />
            </>
          )}
        </Box>
      </ScrollView>
    </Box>
  );
};

export default History;
