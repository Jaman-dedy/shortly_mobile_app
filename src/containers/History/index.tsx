import React from 'react';

import HistoryComponent from '@app/components/History';

const History = () => {
  return <HistoryComponent />;
};

export default History;
