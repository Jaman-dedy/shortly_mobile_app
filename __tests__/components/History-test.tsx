import React from 'react';
import renderer from 'react-test-renderer';
import {cleanup} from '@testing-library/react-native';

import History from '../../src/components/History';

import MockThemes from '../../mock/MockTheme';

afterEach(cleanup);
describe('History component', () => {
  test('History component renders correctly with the loading component', () => {
    const mockCallBack = jest.fn();
    const mockData = {
      loading: true,
      result: [
        {code: '8djdii', link: 'wwww.example.com'},
        {code: '8djdiid', link: 'wwww.example.com'},
        {code: '8dj4dii', link: 'wwww.example.com'},
      ],
    };
    const tree = renderer
      .create(
        <MockThemes>
          <History
            shortenLinkState={mockData}
            handleDeleteLink={mockCallBack}
          />
        </MockThemes>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('History component renders correctly with box history component', () => {
    const mockCallBack = jest.fn();
    const mockData = {
      loading: false,
      result: [
        {code: '8djdii', link: 'wwww.example.com'},
        {code: '8djdiid', link: 'wwww.example.com'},
        {code: '8dj4dii', link: 'wwww.example.com'},
      ],
    };
    const tree = renderer
      .create(
        <MockThemes>
          <History
            shortenLinkState={mockData}
            handleDeleteLink={mockCallBack}
          />
        </MockThemes>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('History component renders correctly with the empty content component', () => {
    const mockCallBack = jest.fn();
    const mockData = {
      loading: false,
      result: [],
    };
    const tree = renderer
      .create(
        <MockThemes>
          <History
            shortenLinkState={mockData}
            handleDeleteLink={mockCallBack}
          />
        </MockThemes>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
