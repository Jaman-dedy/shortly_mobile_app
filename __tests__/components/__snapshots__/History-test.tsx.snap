// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History component History component renders correctly with box history component 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#F0F1F6",
        "flex": 3,
        "marginTop": 16,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#35323E",
          "fontFamily": "Poppins-Medium",
          "fontSize": 18,
          "textAlign": "center",
        },
      ]
    }
  >
    Your Link History
  </Text>
  <RCTScrollView>
    <View>
      <View
        style={
          Array [
            Object {
              "marginTop": 16,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderRadius": 6,
                "marginBottom": 16,
                "marginLeft": 16,
                "marginRight": 16,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignSelf": "stretch",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "padding": 22,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#35323E",
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 18,
                    "textAlign": "center",
                  },
                ]
              }
            >
              undefined...
            </Text>
            <View
              accessibilityLabel="button"
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              < />
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignSelf": "stretch",
                  "borderBottomColor": "#9E9AA7",
                  "borderBottomWidth": 1,
                },
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "padding": 16,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#2ACFCF",
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 18,
                    "textAlign": "center",
                  },
                ]
              }
            />
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "#2ACFCF",
                  "borderRadiu": 4,
                  "marginTop": 8,
                  "opacity": 1,
                  "paddingHorizontal": 16,
                  "paddingVertical": 8,
                  "width": 280,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontFamily": "Poppins-Medium",
                        "fontSize": 22,
                        "fontWeight": "bold",
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  COPY
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderRadius": 6,
                "marginBottom": 16,
                "marginLeft": 16,
                "marginRight": 16,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignSelf": "stretch",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "padding": 22,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#35323E",
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 18,
                    "textAlign": "center",
                  },
                ]
              }
            >
              undefined...
            </Text>
            <View
              accessibilityLabel="button"
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              < />
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignSelf": "stretch",
                  "borderBottomColor": "#9E9AA7",
                  "borderBottomWidth": 1,
                },
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "padding": 16,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#2ACFCF",
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 18,
                    "textAlign": "center",
                  },
                ]
              }
            />
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "#2ACFCF",
                  "borderRadiu": 4,
                  "marginTop": 8,
                  "opacity": 1,
                  "paddingHorizontal": 16,
                  "paddingVertical": 8,
                  "width": 280,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontFamily": "Poppins-Medium",
                        "fontSize": 22,
                        "fontWeight": "bold",
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  COPY
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderRadius": 6,
                "marginBottom": 16,
                "marginLeft": 16,
                "marginRight": 16,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignSelf": "stretch",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "padding": 22,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#35323E",
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 18,
                    "textAlign": "center",
                  },
                ]
              }
            >
              undefined...
            </Text>
            <View
              accessibilityLabel="button"
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              < />
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignSelf": "stretch",
                  "borderBottomColor": "#9E9AA7",
                  "borderBottomWidth": 1,
                },
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "padding": 16,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#2ACFCF",
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 18,
                    "textAlign": "center",
                  },
                ]
              }
            />
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "#2ACFCF",
                  "borderRadiu": 4,
                  "marginTop": 8,
                  "opacity": 1,
                  "paddingHorizontal": 16,
                  "paddingVertical": 8,
                  "width": 280,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontFamily": "Poppins-Medium",
                        "fontSize": 22,
                        "fontWeight": "bold",
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  COPY
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`History component History component renders correctly with the empty content component 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#F0F1F6",
        "flex": 3,
        "marginTop": 16,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#35323E",
          "fontFamily": "Poppins-Medium",
          "fontSize": 18,
          "textAlign": "center",
        },
      ]
    }
  >
    Your Link History
  </Text>
  <RCTScrollView>
    <View>
      <View
        style={
          Array [
            Object {
              "marginTop": 16,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "padding": 8,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#35323E",
                  "fontFamily": "Poppins-Medium",
                  "fontSize": 18,
                  "textAlign": "center",
                },
              ]
            }
          >
            Sorry, your history is empty!
          </Text>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                },
              ]
            }
          >
            <
              height={600}
              width={250}
            />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#35323E",
                  "fontFamily": "Poppins-Medium",
                  "fontSize": 18,
                  "textAlign": "center",
                },
              ]
            }
          >
            click on the button below to shorten more links
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`History component History component renders correctly with the loading component 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#F0F1F6",
        "flex": 3,
        "marginTop": 16,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#35323E",
          "fontFamily": "Poppins-Medium",
          "fontSize": 18,
          "textAlign": "center",
        },
      ]
    }
  >
    Your Link History
  </Text>
  <RCTScrollView>
    <View>
      <View
        style={
          Array [
            Object {
              "marginTop": 16,
            },
          ]
        }
      >
        <View
          onLayout={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "column",
              }
            }
          >
            <View
              style={Object {}}
            >
              <View
                style={
                  Object {
                    "position": "relative",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "borderRadius": 4,
                        "height": 20,
                        "width": 320,
                      },
                      Object {
                        "backgroundColor": "#E1E9EE",
                        "overflow": "hidden",
                      },
                    ]
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "position": "relative",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "borderRadius": 4,
                        "height": 30,
                        "marginTop": 6,
                        "width": 320,
                      },
                      Object {
                        "backgroundColor": "#E1E9EE",
                        "overflow": "hidden",
                      },
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "column",
              }
            }
          >
            <View
              style={Object {}}
            >
              <View
                style={
                  Object {
                    "position": "relative",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "borderRadius": 4,
                        "height": 20,
                        "width": 320,
                      },
                      Object {
                        "backgroundColor": "#E1E9EE",
                        "overflow": "hidden",
                      },
                    ]
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "position": "relative",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "borderRadius": 4,
                        "height": 30,
                        "marginTop": 6,
                        "width": 320,
                      },
                      Object {
                        "backgroundColor": "#E1E9EE",
                        "overflow": "hidden",
                      },
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
